src/index.ts

```diff
@@ -12,7 +15,10 @@ export = class PortDuplexStream extends Duplex {
     super({ objectMode: true });
     this._port = port;
     this._port.onMessage.addListener((msg: unknown) => this._onMessage(msg));
-    this._port.onDisconnect.addListener(() => this._onDisconnect());
+    this._port.onDisconnect.addListener((portObj) => {
+      const err = portObj?.error || runtime.lastError;
```

Comment:
The type library we're using doesn't appear to recognize the `error` property on the Port object. This is a bug.

The library we're using was actually moved into the `@types` repo. it's now called `@types/webextension-polyfill`. But as you can see, this object remains missing: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/webextension-polyfill/namespaces/runtime.d.ts

We can extend the type ourselves, but it should be reported upstream I think.

Reply:
Reported here: https://github.com/Lusito/webextension-polyfill-ts/issues/65 (they're still using the `webextension-polyfill-ts` GitHub project as the generator for the `@types/webextension-polyfill` types).

Reply:
This was fixed in v0.8.1 of `@types/webextension-polyfill`